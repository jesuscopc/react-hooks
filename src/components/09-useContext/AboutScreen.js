import React, { useContext } from 'react';
import { userContext } from './UserContext';

export const AboutScreen = () => {
  const { user, setUser } = useContext(userContext)
  return (
    <div>
      <h1>AboutSCreen</h1>
      <hr/>

      <pre>
        { JSON.stringify(user, null, 3)}
      </pre>

      <button
        className="btn btn-warning"
        onClick={ () => setUser({})}
      >
        Logout
      </button>
    </div>
  )
}
