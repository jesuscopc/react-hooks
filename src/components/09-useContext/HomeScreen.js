import React, { useContext } from 'react';
import { userContext } from './UserContext';

export const HomeScreen = () => {
  const { user } = useContext(userContext);
  return (
    <div>
      <h1>HomeScreen</h1>
      <hr/>
      <pre className="container">
        { JSON.stringify(user, null, 3)}
      </pre>
    </div>
  )
}
