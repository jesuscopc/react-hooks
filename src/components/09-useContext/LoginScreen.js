import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const LoginScreen = () => {
  const { setUser } = useContext(userContext)
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr/>
      <button
        className="btn btn-primary"
        onClick={ () => setUser({
          id: 123,
          name: 'Fernando'
        })}
      >
        Login
      </button>
    </div>
  )
}
