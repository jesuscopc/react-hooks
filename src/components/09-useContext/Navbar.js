import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <Link to="/" className="navbar-brand">UseContext</Link>
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink exact className="nav-link" to="/">Home</NavLink>
            <NavLink exact className="nav-link" to="/about">About</NavLink>
            <NavLink exact className="nav-link" to="/login">Login</NavLink>
          </div>
        </div>
      </div>
    </nav>
  )
}
