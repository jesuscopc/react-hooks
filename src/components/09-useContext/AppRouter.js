import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { Navbar } from "./Navbar";

export const AppRouter = () => {
  return(
    <Router>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/about" component={ AboutScreen }/>
          <Route exact path="/login" component={ LoginScreen }/>
          <Route exact path="/" component={ HomeScreen }/>
        </Switch>
      </div>
    </Router>
  )
}