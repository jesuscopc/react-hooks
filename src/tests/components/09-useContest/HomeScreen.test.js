import React from 'react';
import { shallow } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { userContext } from '../../../components/09-useContext/UserContext';


describe('Sui Test HomeScreen', () => {
  
  const user = {
    name: 'jesus',
    email: 'jesus@gmail.com'
  }
  const wrapper = shallow(
    <userContext.Provider value={{ 
      user
    }}>
      <HomeScreen />
    </userContext.Provider>
  )
  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
})
