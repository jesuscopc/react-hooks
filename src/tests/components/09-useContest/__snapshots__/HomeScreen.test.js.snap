// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui Test HomeScreen should be render correctly 1`] = `<HomeScreen />`;
